### PR TITLE
[Docs] Update flow discovery.* component examples

### PR DESCRIPTION
**Pull request by spartan0x117**
#### PR Description
Updates the `discovery.*` Flow component examples to make them as full and copy-pasteable as possible.

Also adds to the exporter docs-writing guide the instructions for adding a named header for each example when there are multiple.

Most changes are:
- Fixes to section headers to be `Example` and `Examples` when there is one vs. multiple examples.
- Addition of a scrape and remote write component.
- Making placeholder values consistent.
- Making indentation consistent.

#### PR Checklist

- [ ] CHANGELOG updated
- [X] Documentation added
- [ ] Tests updated

### CODE REVIEW

**Review by clayton-cornell — APPROVED** [review]
LGTM

File: docs/sources/flow/reference/components/discovery.file.md [hunk 1]
```diff
@@ -59,21 +59,37 @@ values.
...
+      basic_auth {
+          username = USERNAME
+          password = PASSWORD
+      }
+  }
 }
 ```
+Replace the following:
+  - `LOKI_PATH`: The URL of the Loki server to send logs to.
+  - `USERNAME`: The username to use for authentication to the Loki API.
```
Comment (mattdurham) [inline docs/sources/flow/reference/components/discovery.file.md:89]: nit: In the above we reference remote_write API but here we say Loki, do we want to change remote_write to Prometheus Compatible?

File: docs/sources/flow/reference/components/discovery.file.md [hunk 2]
```diff
@@ -59,21 +59,37 @@ values.
...
+      url = LOKI_PATH
+      basic_auth {
+          username = USERNAME
+          password = PASSWORD
+      }
+  }
 }
 ```
+Replace the following:
+  - `LOKI_PATH`: The URL of the Loki server to send logs to.
```
Comment (mattdurham) [inline docs/sources/flow/reference/components/discovery.file.md:88]: nit: Naming of PATH vs URL?

**Review by mattdurham — APPROVED** [review]
There are some minor nits but overall looks solid

**Pull request merged**